onnxruntime/core/framework/tunable.h

```diff
@@ -215,9 +215,24 @@ class TunableOp {
     }
   }
 
-  static double Profile(Op<ParamsT>& op, const ParamsT* param) {
+  static double Profile(Op<ParamsT>& op, const ParamsT* param, double min_time, bool early_stop) {
     constexpr const int num_iter = 100;
+    constexpr const int early_stop_num_iter = 10;
```

Comment:
> early_stop_num_iter

Would it be better to express this as a fraction of `num_iter`, like you've done below with `early_stop_factor`? Perhaps like `num_iter / 10` or `num_iter / 20`?

Reply:
Actually I think this should be 
```c++
constexpr const int max_tuning_iterations;  # default to 100
constexpr const int max_tuning_duration_ms;  # default to infinity
```

We then probe run the kernel runtime `approx_duration` during warmup.  Then the actually iteration we want to carry out will be `min(max_tuning_iterations, max_tuning_duration_ms/approx_duration)`